test/util/test_file_io.py

```diff
@@ -1,11 +1,61 @@
+from __future__ import annotations
+
 from pathlib import Path
+from unittest import TestCase
 from unittest.mock import MagicMock, patch
 
 from pytest import raises
 
 from flash_patcher.exception.error_manager import ErrorManager
 from flash_patcher.exception.injection import InjectionError
-from flash_patcher.util.file_io import read_safe, readlines_safe, writelines_safe
+from flash_patcher.util.file_io import FileWritebackManager, \
+    read_safe, readlines_safe, writelines_safe
+
+class FileWritebackManagerSpec (TestCase):
+    @patch('pathlib.Path.open', create=True)
+    def test_rw_safe_success(
+        self: FileWritebackManagerSpec,
+        mock_open: MagicMock,
+    ) -> None:
+        mock_file = MagicMock()
+
+        mock_open.return_value.__enter__.return_value = mock_file
+
+        with open("../test/testdata/DoAction1.as", encoding="utf-8") as file:
```

Comment:
Consider using `Path.open()` rather than `open()`. Maybe also move the path to a constant outside of the test functions.

Reply:
Using `Path.open()` breaks some of the tests, as we are replacing this call with a mock. I will move the filename to a constant though.

Reply:
I don't think I know enough to understand, so I'll trust you.